invenio_records_resources/services/custom_fields/base.py

```diff
@@ -37,10 +39,18 @@ def ui_field(self):
         return self.field
 
 
-class ListBaseCF(BaseCF):
+class BaseListCF(BaseCF):
     """Base Custom Field class."""
 
-    def __init__(self, name, multiple=False, **kwargs):
+    def __init__(self, name, field_cls, multiple=False, **kwargs):
         """Constructor."""
         super().__init__(name, **kwargs)
         self._multiple = multiple
+        self._field_cls = field_cls
+
+    @property
+    def field(self):
+        """Marshmallow field custom fields."""
+        _field = self._field_cls(**self._field_args)
```

Comment:
I think here you could do instead 
```suggestion
        _field = self.__class__(**self._field_args)
```
it should give the actual instance class....

Reply:
but `__class__` would point to `BaseCF` or `BaseListCF`. While in this case we are talking about Marshmallow field classes (e.g. `SanitizeUnicode`)

Reply:
my bad! I was confused... Maybe we can add to the constrcutor's docstring the definition of the parameters, so it's all clear and tidy :)

Reply:
Will be tidy, not sure will ever be "clear" with all naming issues 😂